### PR TITLE
Added created timestamp in input log

### LINKED ISSUE #8279: Feature Request: Filebeat timestamp on docker input
In case of docker input, filebeat attaches the @timestamp field from the docker timestamp. I want to capture the time when filebeat actually read this log along with the docker timestamp in a field, say, @filebeattimestamp. The usecase of this could be as a metric to measure filebeat's latency in pumping logs. 

This is a feature request for appending a filebeat timestamp, the name of which could be configurable for docker input. 

This is a reference to the discussion https://discuss.elastic.co/t/filebeat-timestamp-on-docker-input/147686

### PR DESCRIPTION
**Pull request by anandsinghkunwar**
* Added `meta.created`in all inputs of type `log`
* Closes #8279

### CODE REVIEW

**Comment by anandsinghkunwar** [comment]
Hi can anyone re-trigger my build, it seems one of the (not related to this PR) builds was stalled.

**Comment by exekias** [comment]
Thank you for working on this!

Pinging @ph @urso for opinions. User wants to have `event.created` field for `container` input. Having that it's based on `log`, he added the changes here. I'm wondering if they make sense here, or we should perhaps add a config key to enable this, and disable it by default.

ok to test

**Comment by anandsinghkunwar** [comment]
Hi @exekias I have made changes to add a configurable field `add_event_metadata` that is disabled by default.

**Comment by ph** [comment]
@anandsinghkunwar I like the idea of having the `event.created` to the event, where this value is the time when the event was "created" inside filebeat. With that, you effectively get the latency (when event read and when the event was written on disk, assuming the time is extracted from the raw string or other means).

So I am totally +1 on adding this, but I think it should be **ON** by default. I don't see why we should gate this behind a new option. 

Also, adding a new option has always cost associated with it, maintenance and deprecation.

There are a few things we need to add to this PR so we can merge it:

- Adding tests, test the new field is there.
- Adding a changelog.

**Comment by anandsinghkunwar** [comment]
@ph thanks for the comments. I am not very familiar with the test setup in this repository yet, will explore and try to add unit tests.

**Comment by anandsinghkunwar** [comment]
Hey, I browsed through the tests, couldn't really figure out if the function changed was covered in a unit test. So probably will have to write out a fresh one. Is there any dev guide that can help me understand the testing environment/suite in the project? @ph

**Comment by urso** [comment]
I don't think this fix will work as intended. The docker input sets up a log input, which sets up the reader pipeline. If docker is configured, special Docker reader. The reader is configured in the json reader. The docker reader will parse the timestamp from the log message and set it to the message. This PR assigns the timestamp after it has been parsed.

This is a special setting for docker containers and should not be made available or default for all inputs.

File: filebeat/input/log/harvester.go [hunk 1]
```diff
@@ -360,7 +360,16 @@ func (h *Harvester) onMessage(
...
 
+	// Add meta.created in all messages when `add_event_metadata` is enabled
+	// from Elastic Common Schema Reference:
+	// https://www.elastic.co/guide/en/ecs/current/ecs-event.html#_event_field_details
 	var meta common.MapStr
+	if h.config.AddEventMetadata {
+		meta = common.MapStr{
+			"created": message.Ts,
+		}
+	}
```
Comment (urso) [inline filebeat/input/log/harvester.go]: The `message.TS` here is already the parsed docker container read timestamp. We would need to introduce a 'ReadTS' to `message` in order to track when Filebeat did start reading the message.

**Comment by anandsinghkunwar** [comment]
Hi @urso, thanks for the review. I plan to work on this. Will continue working on this the coming weekend.

**Comment by Xplouder** [comment]
@anandsinghkunwar Any progress? Thank you.

**Comment by anandsinghkunwar** [comment]
@Xplouder hey I have been quite busy lately, haven't been able to get to this. Cannot commit to any timeline as of now. :(

**Pull request closed without merging**